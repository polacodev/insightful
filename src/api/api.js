import axios from 'axios';
import 'dotenv/config'

const apiUrl = process.env.API_URL;
const apiToken = process.env.API_TOKEN;

const api = axios.create({
  baseURL: apiUrl,
  headers: {
    'Authorization': `Bearer ${apiToken}`,
  },
});

export default api;
