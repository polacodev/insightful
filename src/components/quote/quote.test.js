import supertest from 'supertest'
import server from '../../app.js'

import { getQuotes } from './quote.api.js'; // Adjust the path accordingly
import api from '../../api/api.js'; // Assuming this is the path to your api module


const request = supertest(server);

describe("Insightful App", () => {
  afterAll((done) => {
    server.close(done)
  })

  describe("Integration Tests", () => {
    it("GET /               - Success Case", async () => {
      const response = await request
        .get("/")
        .expect(200)
      expect(response.text).toBe("Welcome to Insightful App!")
    })

    it("GET /api/v1/quotes  - Success Case", async () => {
      const response = await request
        .get("/api/v1/quotes")
        .expect(200)
      expect(response._body.message).toBe("Fetched quotes successfully")
      expect(Array.isArray(response._body.quotes)).toBeTruthy()
      expect(response._body.totalItems).toEqual(25)
    })
  })

  describe("Snapshot Tests", () => {
    it("GET /api/v1/quotes  - Success Case", async () => {
      const response = await request
        .get("/api/v1/quotes")
      expect(response._body.quotes).toMatchSnapshot();
    })
  })

  describe('getQuotes Bad Request', () => {
    it('GET /quotes (message) - Unsuccess Case', async () => {
      const res = { status: jest.fn(), json: jest.fn() };
      const next = jest.fn();

      const error = new Error()
      error.message = 'Unsuccessful: GET /api/v1/quotes'
      api.get = jest.fn(() => Promise.reject(error));

      await getQuotes({}, res, next);

      expect(next).toHaveBeenCalledWith(new Error('Unsuccessful: GET /api/v1/quotes'));
    });
    it('GET /quotes (Code)    - Unsuccess Case', async () => {
      const res = { status: jest.fn(), json: jest.fn() };
      const next = jest.fn();

      const error = new Error()
      error.ErrorMessage = "GET /api/v1/quotes "
      error.statusCode = 500
      error.message = 'Unsuccessful: ' + error.statusCode + ' ' + error.ErrorMessage
      api.get = jest.fn(() => Promise.reject(error));

      await getQuotes({}, res, next);

    });
  });
})
