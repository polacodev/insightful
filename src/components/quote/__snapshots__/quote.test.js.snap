// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insightful App Snapshot Tests GET /api/v1/quotes  - Success Case 1`] = `
[
  {
    "author": "Ronald Reagan",
    "author_permalink": "ronald-reagan",
    "body": "The taxpayer - that's someone who works for the federal government but doesn't have to take the civil service examination.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 1,
    "id": 28155,
    "private": false,
    "tags": [
      "government",
      "work",
    ],
    "upvotes_count": 2,
    "url": "https://favqs.com/quotes/ronald-reagan/28155-the-taxpayer--",
  },
  {
    "author": "John Mayer",
    "author_permalink": "john-mayer",
    "body": "Ladies, if you want to know the way to my heart... good spelling and good grammar, good punctuation, capitalize only where you are supposed to capitalize, it's done.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 27937,
    "private": false,
    "tags": [
      "good",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/john-mayer/27937-ladies-if-you-",
  },
  {
    "author": "Joseph Stalin",
    "author_permalink": "joseph-stalin",
    "body": "History shows that there are no invincible armies.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 1,
    "id": 31581,
    "private": false,
    "tags": [
      "history",
    ],
    "upvotes_count": 2,
    "url": "https://favqs.com/quotes/joseph-stalin/31581-history-shows-",
  },
  {
    "author": "Karl Lagerfeld",
    "author_permalink": "karl-lagerfeld",
    "body": "One is never over-dressed or underdressed with a Little Black Dress.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 62851,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/karl-lagerfeld/62851-one-is-never--",
  },
  {
    "author": "Dale Carnegie",
    "author_permalink": "dale-carnegie",
    "body": "Today is life-the only life you are sure of. Make the most of today. Get interested in something. Shake yourself awake. Develop a hobby. Let the winds of enthusiasm sweep through you. Live today with gusto.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 4,
    "id": 38245,
    "private": false,
    "tags": [
      "life",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/dale-carnegie/38245-today-is-life-",
  },
  {
    "author": "Alex Lifeson",
    "author_permalink": "alex-lifeson",
    "body": "The shock of any trauma, I think changes your life. It's more acute in the beginning and after a little time you settle back to what you were. However it leaves an indelible mark on your psyche.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 57256,
    "private": false,
    "tags": [
      "time",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/alex-lifeson/57256-the-shock-of--",
  },
  {
    "author": "Mark Twain",
    "author_permalink": "mark-twain",
    "body": "Never let your schooling interfere with your education.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 63831,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/mark-twain/63831-never-let-you-",
  },
  {
    "author": "Nelson Mandela",
    "author_permalink": "nelson-mandela",
    "body": "Communists have always played an active role in the fight by colonial countries for their freedom, because the short-term objects of Communism would always correspond with the long-term objects of freedom movements.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 0,
    "id": 23515,
    "private": false,
    "tags": [
      "freedom",
    ],
    "upvotes_count": 2,
    "url": "https://favqs.com/quotes/nelson-mandela/23515-communists-ha-",
  },
  {
    "author": "Mason Cooley",
    "author_permalink": "mason-cooley",
    "body": "Excuses change nothing, but make everyone feel better.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 9843,
    "private": false,
    "tags": [
      "change",
    ],
    "upvotes_count": 2,
    "url": "https://favqs.com/quotes/mason-cooley/9843-excuses-change-",
  },
  {
    "author": "Oscar Wilde",
    "author_permalink": "oscar-wilde",
    "body": "A work of art is the unique result of a unique temperament.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 4,
    "id": 4326,
    "private": false,
    "tags": [
      "art",
      "work",
    ],
    "upvotes_count": 2,
    "url": "https://favqs.com/quotes/oscar-wilde/4326-a-work-of-art--",
  },
  {
    "author": "Hannibal",
    "author_permalink": "hannibal",
    "body": "I love it when a plan comes together.",
    "dialogue": false,
    "downvotes_count": 2,
    "favorites_count": 5,
    "id": 294,
    "private": false,
    "source": "The A Team, TV series. ",
    "tags": [
      "tv",
      "a-team",
    ],
    "upvotes_count": 1,
    "url": "https://favqs.com/quotes/hannibal/294-i-love-it-when--",
  },
  {
    "author": "Paulo Coelho",
    "author_permalink": "paulo-coelho",
    "body": "Of course, to have money is just great because you can do what you think is important to you. I always was a rich person because money's not related to happiness.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 29180,
    "private": false,
    "tags": [
      "great",
      "happiness",
      "money",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/paulo-coelho/29180-of-course-to--",
  },
  {
    "author": "Dale Carnegie",
    "author_permalink": "dale-carnegie",
    "body": "Inaction breeds doubt and fear. Action breeds confidence and courage. If you want to conquer fear, do not sit home and think about it. Go out and get busy.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 3,
    "id": 11847,
    "private": false,
    "tags": [
      "courage",
      "fear",
      "home",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/dale-carnegie/11847-inaction-bree-",
  },
  {
    "author": "Rumi",
    "author_permalink": "rumi",
    "body": "It may be that the satisfaction I need depends on my going away, so that when I've gone and come back, I'll find it at home.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 4,
    "id": 32254,
    "private": false,
    "tags": [
      "home",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/rumi/32254-it-may-be-tha-",
  },
  {
    "author": "AV",
    "author_permalink": "av",
    "body": "The best way to create the future is to: Act. Learn. Build. Repeat.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 62515,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/av/62515-the-best-way--",
  },
  {
    "author": "Anonymous",
    "author_permalink": "anonymous",
    "body": "When the light turns green, don't race to the next one.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 544,
    "private": false,
    "tags": [
      "journey",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/anonymous/544-when-the-light--",
  },
  {
    "author": "Bret Victor",
    "author_permalink": "bret-victor",
    "body": "If there is any delay in that feedback loop between thinking of something and seeing it and building on it, then there is this whole world of ideas which will never be.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 3,
    "id": 61762,
    "private": false,
    "source": "http://vimeo.com/36579366#t=8m58s",
    "tags": [
      "programming",
      "feedback",
    ],
    "upvotes_count": 1,
    "url": "https://favqs.com/quotes/bret-victor/61762-if-there-is-a-",
  },
  {
    "author": "Marshall McLuhan",
    "author_permalink": "marshall-mcluhan",
    "body": "Anyone who tries to make a distinction between education and entertainment doesn't know the first thing about either.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 16079,
    "private": false,
    "tags": [
      "education",
    ],
    "upvotes_count": 1,
    "url": "https://favqs.com/quotes/marshall-mcluhan/16079-anyone-who-tr-",
  },
  {
    "author": "Captain Henri",
    "author_permalink": "captain-henri",
    "body": "You walk the way for yourself, only for yourself.",
    "context": "In response to Tom, who said he was walking the Camino de Santiago for his deceased son.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 61862,
    "private": false,
    "source": "The Way.  2010 Movie.  19m25s.",
    "tags": [
      "the-way-movie",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/captain-henri/61862-you-walk-the--",
  },
  {
    "author": "Nag R1",
    "author_permalink": "nag-r1",
    "body": "Be kind whenever possible. It is always possible.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 63343,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/nag-r1/63343-be-kind-whene-",
  },
  {
    "author": "Mark Twain",
    "author_permalink": "mark-twain",
    "body": "Never let your schooling interfere with your education.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 2,
    "id": 63203,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/mark-twain/63203-never-let-you-",
  },
  {
    "author": "R. Buckminster Fuller",
    "author_permalink": "r-buckminster-fuller",
    "body": "Nature is trying very hard to make us succeed, but nature does not depend on us. We are not the only experiment.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 3,
    "id": 44698,
    "private": false,
    "tags": [
      "nature",
    ],
    "upvotes_count": 1,
    "url": "https://favqs.com/quotes/r-buckminster-fuller/44698-nature-is-try-",
  },
  {
    "author": "Coco Chanel",
    "author_permalink": "coco-chanel",
    "body": "Women must tell men always that they are the strong ones. They are the big, the strong, the wonderful. In truth, women are the strong ones. It is just my opinion, I am not a professor.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 3,
    "id": 40296,
    "private": false,
    "tags": [
      "men",
      "truth",
      "women",
    ],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/coco-chanel/40296-women-must-te-",
  },
  {
    "author": "Benjamin Disraeli",
    "author_permalink": "benjamin-disraeli",
    "body": "Like all great travellers, I have seen more than I remember, and remember more than I have seen.",
    "dialogue": false,
    "downvotes_count": 0,
    "favorites_count": 3,
    "id": 29387,
    "private": false,
    "tags": [
      "great",
      "travel",
    ],
    "upvotes_count": 1,
    "url": "https://favqs.com/quotes/benjamin-disraeli/29387-like-all-grea-",
  },
  {
    "author": "Author",
    "author_permalink": "author",
    "body": "Your quote",
    "dialogue": false,
    "downvotes_count": 1,
    "favorites_count": 3,
    "id": 63723,
    "private": false,
    "tags": [],
    "upvotes_count": 0,
    "url": "https://favqs.com/quotes/author/63723-your-quote",
  },
]
`;
